frontend/app/routes/banner.tsx

```diff
@@ -298,6 +298,23 @@ export default function Banner() {
     toolActions.setModal(undefined)
   }
 
+  const handleRefresh = (section: 'content' | 'appearance') => {
+    const savedConfig = toolState.savedConfigurations[toolState.activeVersion]
+    if (!savedConfig) return
+
+    const { content, appearance } = splitConfigProperties(savedConfig)
+    toolActions.setToolConfig(
+      section === 'content'
+        ? {
+            ...content,
+            walletAddress: savedConfig.walletAddress
+          }
+        : {
+            ...appearance,
+            walletAddress: savedConfig.walletAddress
+          }
```

Comment:
This maybe clearer:

```ts
let configToUpdate;
if (section === 'content') {
  configToUpdate = { ...content }
}
// ...
```

Also, why do we have to set `walletAddress` separately? And are there any other keys that may need manual setting like this?